lib/mocha/expectation.rb

```diff
@@ -224,6 +225,44 @@ def with(*expected_parameters, &matching_block)
       self
     end
 
+    # Modifies expectation so that the expected method must be called with a block.
+    #
+    # @return [Expectation] the same expectation, thereby allowing invocations of other {Expectation} methods to be chained.
+    #
+    # @example Expected method must be called with a block.
+    #   object = mock()
+    #   object.expects(:expected_method).with_block_given
+    #   object.expected_method { 1 + 1 }
+    #   # => verify succeeds
+    #
+    #   object = mock()
+    #   object.expects(:expected_method).with_block_given
+    #   object.expected_method
+    #   # => verify fails
+    def with_block_given
```

Comment:
Would `with_a_block` and `without_a_block` be better names?

Reply:
I went backward and forwards on the naming of these methods. I quite like the fact that the names I've chosen mirror the naming of the `Kernel#block_given?` method. However, I'll give your suggestion some thought.

Reply:
I've talked to a couple of other people to ask their opinion on this naming and I've decided to stick with the current naming, i.e. `with_block_given` & `with_no_block_given`.